src/main/java/com/google/cloud/teleport/templates/PubSubToSplunk.java

```diff
@@ -363,4 +364,16 @@ public void processElement(ProcessContext context) {
                   }));
     }
   }
+  
+  /**
+   * Utility method to decrypt a Splunk HEC token.
+   *
+   * @param unencryptedToken The Splunk HEC token as a Base64 encoded {@link String} encrypted with a Cloud KMS Key.
+   * @param kmsKey The Cloud KMS Encryption Key to decrypt the Splunk HEC token.
+   * @return Decrypted Splunk HEC token.
+   */
+  private static ValueProvider<String> maybeDecrypt(
+      ValueProvider<String> unencryptedToken, ValueProvider<String> kmsKey) {
+    return new KMSEncryptedNestedValueProvider(unencryptedToken, kmsKey);
```

Comment:
thx @melbrodrigues for this update.
Shouldn't this parameter be named `encryptedToken`?

Reply:
Good catch @rarsan - I think the value may or may-not-be encrypted tbh since this is backwards compatible. The user could choose to send the hec token without any encryption.

Since this is a private method, we can correct it post merge.